### PR TITLE
Overhaul constants

### LINKED ISSUE #2104: Overhaul & extension of constants
We have constants in different locations, e.g.

* `telegram.constants`
* class attributes of `telegram.ParseMode`
* class attributes of `telegram.Dice`

That should be unified, i.e.:

- [x] Gather all constants in `telegram.constants` (I guess a regex-search for `SCREAMING_SNAIL_CASE` should do the trick …)
- [x] Repace the class attributes by links to the constant from `telegram.constant`, (e.g. `ParseMode.HTML = constants.PARSE_MODE_HTML`)
- [x] Add additional links, where suitable (e.g. `Message.MAX_LENGTH = constants.MAX_MESSAGE_LENGTH`)

Moreover, in light of the recent TG update, we should add

- [ ] constants for user ID `777000` and `1087968824`, which act as senders for forwarded channel posts and messages from anonymous admins (see [here](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Code-snippets#exclude-messages-from-anonymous-admins))
- [ ] Add properties `is_anonymous_admin` and `is_service_chat` (or similar namings) to `User` and `Chat` that compare the user/chat_id to the IDs menotioned above

*Edit:* there should be notes on the constants, if they are not documented. And going through the docs to see if there are constants not covered already probably won't hurt …

### PR DESCRIPTION
**Pull request by NikitaPirate**
closes #2104

### CODE REVIEW

File: telegram/chat.py [hunk 1]
```diff
@@ -92,14 +92,14 @@ class Chat(TelegramObject):
...
-    PRIVATE: str = 'private'
-    """:obj:`str`: 'private'"""
-    GROUP: str = 'group'
-    """:obj:`str`: 'group'"""
-    SUPERGROUP: str = 'supergroup'
-    """:obj:`str`: 'supergroup'"""
-    CHANNEL: str = 'channel'
-    """:obj:`str`: 'channel'"""
+    PRIVATE: str = constants.CHAT_PRIVATE
+    """:const:`telegram.constants.CHAT_PRIVATE`"""
```
Comment (Bibo-Joshi) [inline telegram/chat.py:96]: Did you build the docs to check that this renders correctly?

File: telegram/constants.py [hunk 2]
```diff
@@ -39,6 +39,110 @@
     MAX_MESSAGE_ENTITIES (:obj:`int`): 100 (Beyond this cap telegram will simply ignore further
         formatting styles)
 
+The following constants are attributes of classes:
```
Comment (Bibo-Joshi) [inline telegram/constants.py:42]:
```suggestion
The following constants are related to specific classes and are also available as attributes of those classes:
```

File: telegram/constants.py [hunk 3]
```diff
@@ -39,6 +39,110 @@
...
+
+:class:`telegram.Poll`:
+
+Attributes:
+    POLL_REGULAR (:obj:`str`): 'regular'
+    POLL_QUIZ (:obj:`str`): 'quiz'
+
+
+
+:class:`telegram.ext.ConversationHandler`:
```
Comment (Bibo-Joshi) [inline telegram/constants.py:117]: Upsie :D I didn't think of the fact that we have class variables in `telegram.ext/utils.` Those shouldn't go here, though. Basically, we're just concerned with the constants of TG classes. Please revert :)

File: telegram/files/inputfile.py [hunk 4]
```diff
@@ -24,11 +24,12 @@
 import os
 from uuid import uuid4
 
-from telegram import TelegramError
+from telegram import TelegramError, constants
 
 from typing import IO, Tuple, Optional
 
-DEFAULT_MIME_TYPE = 'application/octet-stream'
+DEFAULT_MIME_TYPE = constants.INPUTFILE_DEFAULT_MIME_TYPE
```
Comment (Bibo-Joshi) [inline telegram/files/inputfile.py:31]: I think we should skip this one too. I didn't have a doc string for reason, I think …

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Thanks for the PR!

A left a few comments. Also in #2104 I mentioned new constants & properties for anonymous admins & servce chat. Could you add those, too? But we can also do that in a spearate PR.

Regarding Codacy: Don't worry about it, I don't know why it complains about the docstrings … maybe

File: telegram/chat.py (hunk 1 above)
Comment (NikitaPirate) [reply telegram/chat.py:96]: Yes I built. Everything is displayed correctly and links work.

Comment (Bibo-Joshi) [reply telegram/chat.py:96]: 👍

File: telegram/messageentity.py [hunk 5]
```diff
@@ -90,36 +90,36 @@ def de_json(cls, data: Optional[JSONDict], bot: 'Bot') -> Optional['MessageEntit
...
+    """:const:`telegram.constants.MESSAGEENTITY_PRE`"""
+    TEXT_LINK: str = constants.MESSAGEENTITY_TEXT_LINK
+    """:const:`telegram.constants.MESSAGEENTITY_TEXT_LINK`"""
+    TEXT_MENTION: str = constants.MESSAGEENTITY_TEXT_MENTION
+    """:const:`telegram.constants.MESSAGEENTITY_TEXT_MENTION`"""
+    UNDERLINE: str = constants.MESSAGEENTITY_UNDERLINE
+    """:const:`telegram.constants.MESSAGEENTITY_UNDERLINE`"""
+    STRIKETHROUGH: str = constants.MESSAGEENTITY_STRIKETHROUGH
+    """:const:`telegram.constants.MESSAGEENTITY_STRIKETHROUGH`"""
     ALL_TYPES: List[str] = [
```
Comment (Bibo-Joshi) [inline telegram/messageentity.py:123]: Can you move this, too? Overlooked in the last review

File: telegram/ext/conversationhandler.py [hunk 6]
```diff
@@ -168,7 +168,6 @@ class ConversationHandler(Handler):
         ValueError
 
     """
-
```
Comment (Bibo-Joshi) [inline telegram/ext/conversationhandler.py:171]: This one's probably left from the previous changes. Can you remove it? Just looks confusing in the git log

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Thanks for the update :) Just two minor comments left.

PS: You can leave the PR title as is. GH pings me on updates ;)

File: telegram/messageentity.py (hunk 5 above)
Comment (NikitaPirate) [reply telegram/messageentity.py:123]: Remove it? Or what? I don't understand.

Comment (Bibo-Joshi) [reply telegram/messageentity.py:123]: I mean move `ALL_TYPES` to `tg.constants`, too

Comment (NikitaPirate) [reply telegram/messageentity.py:123]: And ALL_EMOJI in Dice too, right?

Comment (Bibo-Joshi) [reply telegram/messageentity.py:123]: uh, yes :D good catch!

**Review by Bibo-Joshi — APPROVED** [review]
Thanks for the contribution :) I'll wait for the tests to run and merge

**Pull request merged**